server/src/main/java/org/elasticsearch/action/admin/indices/stats/CommonStats.java

```diff
@@ -125,42 +133,70 @@ public CommonStats(CommonStatsFlags flags) {
-                    case Translog -> translog = indexShard.translogStats();
-                    case RequestCache -> requestCache = indexShard.requestCache().stats();
-                    case Recovery -> recoveryStats = indexShard.recoveryStats();
-                    case Bulk -> bulk = indexShard.bulkStats();
+                    case Docs -> stats.docs = indexShard.docStats();
+                    case Store -> stats.store = indexShard.storeStats();
+                    case Indexing -> stats.indexing = indexShard.indexingStats();
+                    case Get -> stats.get = indexShard.getStats();
+                    case Search -> stats.search = indexShard.searchStats(flags.groups());
+                    case Merge -> stats.merge = indexShard.mergeStats();
+                    case Refresh -> stats.refresh = indexShard.refreshStats();
+                    case Flush -> stats.flush = indexShard.flushStats();
+                    case Warmer -> stats.warmer = indexShard.warmerStats();
+                    case QueryCache -> stats.queryCache = indicesQueryCache.getStats(indexShard.shardId());
+                    case FieldData -> stats.fieldData = indexShard.fieldDataStats(flags.fieldDataFields());
+                    case Completion -> stats.completion = indexShard.completionStats(flags.completionDataFields());
+                    case Segments -> stats.segments = indexShard.segmentStats(
+                        flags.includeSegmentFileSizes(),
+                        flags.includeUnloadedSegments()
+                    );
+                    case Translog -> stats.translog = indexShard.translogStats();
+                    case RequestCache -> stats.requestCache = indexShard.requestCache().stats();
+                    case Recovery -> stats.recoveryStats = indexShard.recoveryStats();
+                    case Bulk -> stats.bulk = indexShard.bulkStats();
                     case Shards ->
                         // Setting to 1 because the single IndexShard passed to this method implies 1 shard
-                        shards = new ShardCountStats(1);
-                    default -> throw new IllegalStateException("Unknown Flag: " + flag);
+                        stats.shards = new ShardCountStats(1);
+                    default -> throw new IllegalStateException("Unknown or invalid flag for shard-level stats: " + flag);
                 }
             } catch (AlreadyClosedException e) {
                 // shard is closed - no stats is fine
             }
         }
+
+        return stats;
+    }
+
+    /**
+     * Filters the given flags for {@link CommonStatsFlags#INDEX_LEVEL} flags and calculates the corresponding statistics.
+     */
+    public static CommonStats getIndexLevelStats(final IndexService indexService, CommonStatsFlags flags) {
+        // Filter index level flags (currently it is just the Mappings flag, so we just filter with an if statement).
+        if (flags.isSet(CommonStatsFlags.Flag.Mappings)) {
```

Comment:
There's actually no need to be this complicated here I think. We could just do:

```java
var stats = new CommonStats(CommonStatsFlags.NONE);
if (flags.isSet(CommonStatsFlags.Flag.Mappings)) {
   stats.nodeMappings = indexService.getNodeMappingStats();
}
return stats;
```

couldn't we?

Reply:
Yes we can do that. In next commits.